src/os_common/os_fstream.c

```diff
@@ -134,8 +134,12 @@ __fstream_printf(WT_SESSION_IMPL *session, WT_FSTREAM *fstr, const char *fmt, va
         va_copy(ap_copy, ap);
         if ((p = buf->mem) != NULL)
             p += buf->size;
-        WT_ASSERT(session, buf->memsize >= buf->size);
         space = buf->memsize - buf->size;
```

Comment:
Moving the `WT_ASSERT` to after `space = buf->memsize - buf->size;` makes it possible for `space` to underflow even though we'll still catch it on the next line. I'm going to add our UBSan tests to this PR to make sure UBSan doesn't complain about this.

Reply:
ah that's a good point, i didn't think about tripping UBSan even though it's handled. thanks!

Reply:
hmm, on second thoughts - UBSan should be fine, these are both unsigned types so underflow is not undefined behaviour.

Reply:
Ah, ok. No worries then and UBSan's saying the same.